pandas/tests/io/test_common.py

```diff
@@ -326,7 +346,8 @@ def test_next(self, mmap_file):
             next_line = next(wrapper)
             assert next_line.strip() == line.strip()
 
-        pytest.raises(StopIteration, next, wrapper)
+        with pytest.raises(StopIteration, match=r'$^'):
+            next(wrapper)
```

Comment:
I'm surprised `r'$^'` works, but it seems it does. In any case, it's a bit confusing to read, because normally the caret (`^`) checks for the beginning of a line and the dollar for the end. I'd switch those characters around.

Reply:
@h-vetinari well spotted. thanks. i'll get that sorted.